### ISSUE TITLE
[BUG] KMS encryption of static-website bucket preventing cloudfront distribution.

### ISSUE THREAD

**Issue by JoshuaToth** [aws/aws-pdk#45]
### Describe the bug

With the addition of a default KMS encryption on the website bucket, cloudfront is now unable to serve the files:

### Expected Behavior

Cloudfront should be able to serve the website files through it's distribution. 

### Current Behavior

![image](https://user-images.githubusercontent.com/10456811/174531762-42e57ef2-efe8-45c6-83db-db9585ea3dfb.png)

### Reproduction Steps

Not really any sort of extra config outside of the default: 

```typescript
    new StaticWebsite(this, "StaticFrontend", {
      websiteContentPath: path.join(
        __dirname,
        "../../../frontend/build"
      ),
      runtimeOptions: {
        jsonPayload: {
          identityPoolId: userIdentity.identityPool.identityPoolId,
          userPoolId: userIdentity.userPool?.userPoolId,
          userPoolClientId: userIdentity.userPoolClient?.userPoolClientId,
          region: stack.region,
        },
      },
    });
```

The files _are_ in s3

### Possible Solution

It looks like a common fix for this is to provide a lambda@edge function to help with retrieving the files. Seems like overkill though...

### Additional Information/Context

This is a fresh deployment with the latest version: 0.2.1

### PDK version used

static-website: 0.2.1

### What languages are you seeing this issue on?

Typescript

### Environment details (OS name and version, etc.)

OSX 12.3.1, deployed from local using CDK

**Comment by cogwirrel** [comment]
S3 OAI doesn't seem to support KMS out of the box so that lambda approach might be the only way to go: https://github.com/aws/aws-cdk/blob/cea1039e3664fdfa89c6f00cdaeb1a0185a12678/packages/%40aws-cdk/aws-cloudfront-origins/lib/s3-origin.ts#L74-L82

**Comment by agdimech** [comment]
Good first issue Josh. 

Commencing investigation.

**Comment by agdimech** [comment]
It does appear OAI does not support KMS encryption without a Lambda@Edge signer.

Posting a PR to change default encryption to S3_MANAGED as using KMS would only be mandatory for regulated customers. If they require KMS, the current StaticWebsite API allows them to associate a Lamda@Edge function to the Cloudfront default behaviour and they have the option of either passing in their own S3 bucket with KMS encryption or using the newly added KMS encryption type.

I have raised #49 to potentially add support for zero touch support for KMS encryption in the future.

**Comment by agdimech** [comment]
Code is merged and should be available in PDK version > 0.2.3 once released (ETA: 11:30 am 21/6)

**Comment by JoshuaToth** [comment]
Thanks @agdimech for the quick turnaround. The new version works great. 💯
